### PR TITLE
Remove login/logout shellout in favor of providing config file

### PR DESCRIPTION
**Pull request by elliott-davis**
tenor-248321945
This has the benefit of not destroying any existing docker config settings you might have on your system

Signed-off-by: Travis Elliott Davis <edavis@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/pkg-export-docker/src/lib.rs [hunk 1]
```diff
@@ -139,23 +138,15 @@ impl Credentials {
...
             }
             _ => {
                 Ok(Credentials {
-                    username: username.to_string(),
-                    password: password.to_string(),
+                    token: base64::encode(&format!(
+                        "{}:{}",
+                        username.to_string(),
+                        password.to_string()
+                    )),
```
Comment (christophermaier) [inline components/pkg-export-docker/src/lib.rs]: This is really the token?

The match on `_` makes me a bit nervous... is there a reason to use a string type here, as opposed to an Enum?

Comment (elliott-davis) [reply components/pkg-export-docker/src/lib.rs]: yup that's really _really_ the token. Ain't no security like base64 security

Comment (christophermaier) [reply components/pkg-export-docker/src/lib.rs]: tenor-111366692

Comment (elliott-davis) [reply components/pkg-export-docker/src/lib.rs]: I guess you could accomplish this with an enum if one of the values was `default`. The idea behind this was you can have `amazon` and everything else defaulted to `docker` but maybe that's not the best UX.

Comment (christophermaier) [reply components/pkg-export-docker/src/lib.rs]: The inputs will be driven by the UI, and thus the API, right? We can have an enum that's got `Amazon` and `Docker` values, with an accompanying `FromStr` implementation that lets us parse input into the enum values. That way we offload some cognitive work to the compiler and make sure we don't leave any holes when adding new export destinations.

**Comment by christophermaier** [comment]
tenor-91011069

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/313562474).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/313562474).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**